src/common/pico_audio/include/pico/audio.h

```diff
@@ -24,13 +24,21 @@ extern "C" {
 
 // PICO_CONFIG: SPINLOCK_ID_AUDIO_FREE_LIST_LOCK, Spinlock number for the audio free list, min=0, max=31, default=6, group=audio
 #ifndef SPINLOCK_ID_AUDIO_FREE_LIST_LOCK
+#if PICO_RP2350 && !PICO_USE_SW_SPIN_LOCKS
+#error "When not using software spin locks on RP2350, you must explicitly define the pico-extras spinlock IDs, accounting for Errata E2 and the SDK spin lock IDs"
+#else
 #define SPINLOCK_ID_AUDIO_FREE_LIST_LOCK 6
 #endif
+#endif
 
 // PICO_CONFIG: SPINLOCK_ID_AUDIO_PREPARED_LISTS_LOCK, Spinlock number for the audio prepared list, min=0, max=31, default=7, group=audio
 #ifndef SPINLOCK_ID_AUDIO_PREPARED_LISTS_LOCK
+#if PICO_RP2350 && !PICO_USE_SW_SPIN_LOCKS
+#error "When not using software spin locks on RP2350, you must explicitly define the pico-extras spinlock IDs, accounting for Errata E2 and the SDK spin lock IDs"
```

Comment:
Rather than using the same generic error message for each of these, I wonder if it might be more useful if it said e.g.
```suggestion
#error "When not using software spin locks on RP2350, you must explicitly define the SPINLOCK_ID_AUDIO_PREPARED_LISTS_LOCK, taking Errata RP2350-E2 and the SDK spin lock IDs into account"
```

Reply:
```suggestion
#error "When not using software spin locks on RP2350, you must explicitly define SPINLOCK_ID_AUDIO_PREPARED_LISTS_LOCK, taking Errata RP2350-E2 and the SDK spin lock IDs into account"
```